Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -9015,6 +9015,226 @@ def inner(client, args):
+            latest_incident_time = incident_time
+
+    if increase_last_run_time:
+        latest_incident_time = latest_incident_time + timedelta(milliseconds=1)
+
+    return latest_incident_time.strftime(date_format)
+
+
+def remove_old_incidents_ids(found_incidents_ids, current_time, look_back):
+    """
+    Removes old incident ids
+
+    :type found_incidents_ids: ``dict``
+    :param found_incidents_ids: Dict of incidents ids
+
+    :type current_time: ``int``
+    :param current_time: The current epoch time
+
+    :type look_back: ``int``
+    :param look_back: The look back time in minutes
+
+    :return: The new incidents ids
+    :rtype: ``dict``
+    """
+    look_back_in_seconds = look_back * 60
+    deletion_threshold_in_seconds = look_back_in_seconds * 2
+
+    new_found_incidents_ids = {}
+    for inc_id, addition_time in found_incidents_ids.items():
+
+        if current_time - addition_time < deletion_threshold_in_seconds:
+            new_found_incidents_ids[inc_id] = addition_time
+
+    return new_found_incidents_ids
+
+
+def set_next_fetch_run(last_run, incidents, fetch_limit, start_fetch_time, end_fetch_time, look_back,
+                       created_time_field, id_field='id', date_format='%Y-%m-%dT%H:%M:%S', save_incidents_in_last_run=False,
+                       increase_last_run_time=False):
+    """
+    Sets the next run
+
+    :type last_run: ``dict``
+    :param last_run: The LastRun object
+
+    :type incidents: ``list``
+    :param incidents: List of incidents
+
+    :type fetch_limit: ``int``
+    :param fetch_limit: The fetch limit
+
+    :type start_fetch_time: ``str``
+    :param start_fetch_time: The start time to fetch
+
+    :type end_fetch_time: ``str``
+    :param end_fetch_time: The end time to fetch
+
+    :type look_back: ``int``
+    :param look_back: The time to look back in fetch in minutes
+
+    :type created_time_field: ``str``
+    :param created_time_field: The incident created time field
+
+    :type id_field: ``str``
+    :param id_field: The incident id field
+
+    :type date_format: ``str``
+    :param date_format: The date format
+
+    :type save_incidents_in_last_run: ``bool``
+    :param save_incidents_in_last_run: Whether to incidents in the last run object
+
+    :type increase_last_run_time: ``bool``
+    :param increase_last_run_time: Whether to increase the last run time with one millisecond
+
+    :return: The new last run object and list of incidents
+    :rtype: ``Tuple``
+    """
+    found_incidents = last_run.get('found_incident_ids', {})
+    current_time = int(time.time())
+
+    incidents_from_limit = incidents[fetch_limit:]
+    incidents = incidents[:fetch_limit]
+
+    for incident in incidents:
+        found_incidents[incident[id_field]] = current_time
+
+    found_incidents = remove_old_incidents_ids(found_incidents, current_time, look_back)
+
+    if len(incidents) == 0:
+        new_last_run = {
+            'time': end_fetch_time,
```

Comment:
best if we set each field like this:
```suggestion
        last_run['time'] = end_fetch_time
```
instead of re-defining the whole last run object, since there could be additional fields it holds. please handle the following lines as well.

another option:
```
last_run.update(new_last_run)
```
but eventually, you must return the last_run obj.

Reply:
I prefer to not do any act on the existing lastRun object in the CommonServerPython file:

1. For getting the developer both lastRun objects (before and after he called the function) for possible other actions he may want to do in the integration itself.
2. As I understood from Anar, we don't want to make in the CSP file any actions related to the integration (might be also because reason number 1).